src/main/java/codeu/model/data/Activity.java

```diff
@@ -0,0 +1,100 @@
+package codeu.model.data;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.UUID;
+
+/**
+ * Class representing an activity (creating user, message, or conversation). Any creation done by a
+ * user prompts the creation of an activity object.
+ */
+public class Activity {
+  public final UUID id;
+  public final UUID owner;
+  public final Action action;
+  public final Instant creation;
+  public final String thumbnail;
+
+  /**
+   * Constructs a new activity.
+   *
+   * @param id the ID of this Activity
+   * @param owner the ID of the owner of this activity
+   * @param action the action of the activity (join, send, create)
+   * @param creation the creation time of this Conversation
+   * @param thumbnail a short summary about the activity
+   */
+  public Activity(UUID id, UUID owner, Action action, Instant creation, String thumbnail) {
+    this.id = id;
+    this.action = action;
+    this.owner = owner;
+    this.creation = creation;
+    this.thumbnail = thumbnail;
+  }
+
+  public Activity(User u) {
+    this(
+        u.getId(),
+        u.getId(),
```

Comment:
Why do we need two of the same IDs?

Reply:
The activity's id and owner's id are the same because this is a registering action. (both the activity and the owner's ID are the same)

Reply:
I see. I am a bit iffy about the way it's displayed. I think it looks a bit confusing with both showing. Maybe we can add a comment?

Reply:
Ok, the other PR fixed this with a comment.